### PR TITLE
[RFR] A service provide data for server or for client with the same call

### PR DESCRIPTION
**Pull request by JulienMattiussi**
Call of symfony API are only in gameservice.
Functions of gameservice can be called from server or client side.

### CODE REVIEW

File: config/config.dist.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 export default {
-    apiUrl: 'localhost/gameapi',
+    //apiUrl: 'localhost/gameapi',
```
Comment (Kmaschta) [inline config/config.dist.js]: Debug found

File: server.js [hunk 2]
```diff
@@ -13,15 +13,21 @@ const dev = process.env.NODE_ENV !== 'production';
...
-        listenedGame[idGame] = setInterval(() => {
-            let url = `http://${config.apiUrl}/${idGame}`;
+        listenedGame[idGame] = setInterval(async () => {
+            let game = await getGame(idGame);
+            if (gameListeners[idGame])
+                gameListeners[idGame].forEach(s => {
+                    s.emit(`game${idGame}`, game);
+                });
+
+            /* let url = `http://${config.apiUrl}/${idGame}`;
```
Comment (Kmaschta) [inline server.js]: Debug found

**Review by Kmaschta — CHANGES_REQUESTED** [review]

File: pages/Game.js [hunk 3]
```diff
@@ -27,14 +26,11 @@ class Game extends Component {
     };
 
     static async getInitialProps() {
-        const url = `http://${config.apiUrl}/490`;
-
-        const res = await fetch(url);
-        const data = await res.json();
+        const game = await getGame(490);
```
Comment (ThieryMichel) [inline pages/Game.js]: What if we have a 404 error, or no internet ?

**Review by ThieryMichel — CHANGES_REQUESTED** [review]

File: src/services/gameservice.js [hunk 4]
```diff
@@ -1,19 +1,18 @@
...
 
 export const getGame = async idGame => {
     let url = `${config.apiUrl}/${idGame}`;
     return fetch(url)
-        .then(handleErrors)
         .then(res => res.json())
         .then(res => {
             return res;
+        })
+        .catch(function(error) {
```
Comment (Kmaschta) [inline src/services/gameservice.js]: .catch(doNothingOnError) would works as well

File: src/services/gameservice.js [hunk 5]
```diff
@@ -1,19 +1,18 @@
 import fetch from 'isomorphic-unfetch';
 import config from '../../config/config.dist';
 
-function handleErrors(res) {
-    if (!res.ok) {
-        return Promise.reject(res.statusText);
-    }
-    return res;
-}
+const doNothingOnError = error => {
```
Comment (Kmaschta) [inline src/services/gameservice.js]: Same as `const doNothingOnError = error => {}`

File: src/services/gameservice.js [hunk 6]
```diff
@@ -1,15 +1,20 @@
...
-        request(url, function(error, response, body) {
-            if (!error && response.statusCode == 200) {
-                if (!callBack) return body;
-                callBack(args, body);
-            }
+export const getGame = async idGame => {
+    let url = `${config.apiUrl}/${idGame}`;
+    return fetch(url)
+        .then(res => res.json())
+        .then(res => {
```
Comment (ThieryMichel) [inline src/services/gameservice.js]: this do nothing

File: src/services/gameservice.js [hunk 7]
```diff
@@ -1,15 +1,20 @@
...
+            return res;
+        })
+        .catch(function(error) {
```
Comment (ThieryMichel) [inline src/services/gameservice.js]: catch(doNothingOnError)
the wrapping function bring nothing

File: src/services/gameservice.js [hunk 8]
```diff
@@ -1,15 +1,20 @@
-const request = require('request');
+import fetch from 'isomorphic-unfetch';
+import config from '../../config/config.dist';
 
-const config = { apiUrl: 'localhost/gameapi' };
+const doNothingOnError = error => {
```
Comment (ThieryMichel) [inline src/services/gameservice.js]: logError would be better

**Review by ThieryMichel — CHANGES_REQUESTED** [review]

**Review by ThieryMichel — APPROVED** [review]

**Pull request merged**